### PR TITLE
Hero headers into Modals

### PR DESCRIPTION
**Pull request by mperrotti**
#### Related issues
Fixes https://meetup.atlassian.net/browse/WC-35
https://meetup.atlassian.net/browse/MW-814

#### Description
Some modals, like the RSVP success modal, are designed to have a full bleed hero header

#### Screenshots (if applicable)
screen shot 2017-04-21 at 10 17 19 am
screen shot 2017-04-21 at 10 17 26 am

### CODE REVIEW

File: assets/scss/components/_modal.scss [hunk 1]
```diff
@@ -103,6 +104,16 @@ lays out as a dialog on background content on larger screens.
 	z-index: map-get($zindex-map, "modal");
 }
 
+.view--modal {
+	.stripe-heroContent {
```
Comment (mperrotti) [inline assets/scss/components/_modal.scss]: Having to fight `stripe-heroContent`. Is there a better way?

File: assets/scss/components/_stripe.scss [hunk 2]
```diff
@@ -31,6 +31,10 @@ Class                | Description
 	@include responsiveVarContext--base() {
 		padding-bottom: $base*2;
 	}
+
+	&:first-child {
```
Comment (mperrotti) [inline assets/scss/components/_stripe.scss]: Put this in a separate PR?

File: src/interactive/Modal.jsx [hunk 3]
```diff
@@ -129,6 +133,19 @@ class Modal extends React.Component {
...
+			<div className='padding--all'>
+				<Button onClick={this.onDismiss} className={dismissButtonClasses}>
+					<Icon shape='cross' size='s' />
+				</Button>
+			</div>
+		);
+
+		const heroStyles = {
+			backgroundColor: heroBgColor || 'transparent',
+			backgroundImage: `url(${heroBgImage})`
```
Comment (mperrotti) [inline src/interactive/Modal.jsx]: When `heroBgImage` isn't set, `background-image: url(undefined)` gets put in the markup. Is there a neater way?

**Comment by mperrotti** [comment]
Still very much in progress. Just want some feedback before I continue. Added comments

File: src/interactive/modal.story.jsx [hunk 4]
```diff
@@ -39,6 +49,73 @@ storiesOf('Modal', module)
...
+	.addWithInfo(
+		'has hero - color',
+		'This is the component with an extended header.',
+		() => (
+			<div style={wrapperStyle}>
+				<InfoWrapper>
+					<Modal
+						onDismiss={onDismiss}
+						heroInverted
+						heroBgColor='rgb(55,30,172)'
```
Comment (mperrotti) [inline src/interactive/modal.story.jsx]: Should we add JS to what `meetup-swatches` exports?

**Comment by akdetrick** [comment]
the "X" alignment seems off by a bit
c6d93e14-267b-11e7-9689-2dcdce53edb7 copy

**Comment by mmcgahan** [comment]
> the "X" alignment seems off by a bit

I agree - it should be over on the right like _every other modal on the internet_ rather than being the first thing that people see when looking at the content 😸

File: assets/scss/components/_modal.scss (hunk 1 above)
Comment (akdetrick) [inline assets/scss/components/_modal.scss]: There is, but it requires more work that doesn't belong in this PR. I filed a ticket for a separate cleanup task:
https://meetup.atlassian.net/browse/SDS-253

File: assets/scss/components/_stripe.scss (hunk 2 above)
Comment (akdetrick) [inline assets/scss/components/_stripe.scss]: nah

File: src/interactive/Modal.jsx (hunk 3 above)
Comment (mmcgahan) [inline src/interactive/Modal.jsx]:
```js
backgroundImage: heroBgImage && `url(${heroBgImage})`,
```

File: src/interactive/Modal.jsx [hunk 5]
```diff
@@ -99,6 +100,10 @@ class Modal extends React.Component {
 			className,
 			children,
 			fullscreen,
+			heroBgColor,
+			heroBgImage,
+			heroContent,
+			heroInverted,
```
Comment (akdetrick) [inline src/interactive/Modal.jsx]: I like all the other prop names, but for inverted context, I'd stick with `inverted` instead of `heroInverted`. The `inverted` prop is present on a number of components and it would be good to keep it consistent.

File: src/interactive/Modal.jsx (hunk 3 above)
Comment (akdetrick) [inline src/interactive/Modal.jsx]:
```
backgroundImage: heroBgImage && `background-image: url(${heroBgImage})`
```

File: src/interactive/Modal.jsx [hunk 6]
```diff
@@ -129,6 +133,19 @@ class Modal extends React.Component {
 			</div>
 		);
 
+		const closeArea = (
+			<div className='padding--all'>
```
Comment (akdetrick) [inline src/interactive/Modal.jsx]: this could solve the left alignment problem with the close button in a hero (pass an arg `hasHero` or something)

```
const closeArea = hasHero => (
	<div className={cx(
		'padding--all',
		{ 'flush--left': hasHero }
	)}>
	...
);
```

File: src/interactive/modal.story.jsx (hunk 4 above)
Comment (akdetrick) [inline src/interactive/modal.story.jsx]: We have an open ticket for this. It's not an urgent need, but it would definitely be nice to have later.

File: src/interactive/Modal.jsx (hunk 3 above)
Comment (akdetrick) [inline src/interactive/Modal.jsx]: what mike said

**Comment by mperrotti** [comment]
> the "X" alignment seems off by a bit

@akdetrick and @mmcgahan 
This should be fixed by a recent change to `button--reset`. Just have to update the `swarm-sasstools` package number

File: src/interactive/modal.test.js [hunk 7]
```diff
@@ -72,6 +76,55 @@ describe('Modal', () => {
 	});
 });
 
+describe('Modal hero header', () => {
+	let modal, heroEl;
+	const content = 'Test model content';
+	const bgColor = 'rgb(55, 30, 172)';
+	const bgImage = 'http://www.cheatsheet.com/wp-content/uploads/2016/09/Homemade-Meat-Gyro-with-French-Fries.jpg';
```
Comment (akdetrick) [inline src/interactive/modal.test.js]: `Homemade-Meat-Gyro-with-French-Fries.jpg` :1st_place_medal:
